src/main/java/de/ait/secondlife/services/CategoryServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+package de.ait.secondlife.services;
+
+import de.ait.secondlife.dto.CategoryDto;
+import de.ait.secondlife.dto.IsActiveCategoryDto;
+import de.ait.secondlife.models.Category;
+import de.ait.secondlife.repositories.CategoriesRepository;
+import de.ait.secondlife.services.interfaces.CategoryService;
+import de.ait.secondlife.services.mapping.CategoryMappingService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+
+    private final CategoriesRepository repository;
+
+    private final CategoryMappingService mappingService;
+
+    @Override
+    public CategoryDto getById(Long id) {
+
+        if (id == null || id <1){
+            throw new RuntimeException("Category ID is incorrect");
+        }
+
+        Category category = repository.findById(id).orElse(null);
+
+        if (category == null) {
+            throw new RuntimeException("Category not found");
+        }
+
```

Comment:
you can use .orElseThrow(()->  RuntimeException("Category not found"))  instead of  .orElse(null);

        if (category == null) {
            throw new RuntimeException("Category not found");
        }

Reply:
Done